sbin/spark-daemon.sh

```diff
@@ -78,6 +78,12 @@ spark_rotate_log ()
     if [ -n "$2" ]; then
 	num=$2
     fi
+
+    expr $num + 0 &>/dev/null
+    if [ $? -ne 0 ] || [ $num -eq 0 ]; then
```

Comment:
@srowen 
Yes, we should avoid this scenario that num isn't number.If num isn't a number,num will be set to 5.